src/accsr/remote_storage.py

```diff
@@ -30,6 +31,19 @@ def download(
         pass
 
 
+class LocalObject(RemoteObjectProtocol):
+    def download(
```

Comment:
The local object cannot have a reasonable definition for download, so it shouldn't form part of its interface. Later we can unify these interfaces but for now the easiest would be to drop the inheritance and to remove the download method

Reply:
Yes, this is again because of the return type of the push/pull functions. Created ObjectProtocol(Protocol) class and let RemoteObjectProtocol and LocalObject inherit from it. Changed the return type List[ObjectProtocol].